src/Diagnostics/FxCop/Desktop.Analyzers/VisualBasic/MSInternal/DoNotCatchCorruptedStateExceptionsAnalyzer.vb

```diff
@@ -0,0 +1,122 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Collections.Immutable
+
+Imports Microsoft.CodeAnalysis
+Imports Microsoft.CodeAnalysis.Diagnostics
+Imports Microsoft.CodeAnalysis.VisualBasic
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Imports SDL.Analyzer.Common
+
+
+Namespace Desktop.Analyzers
+
+    ''' <summary>
+    ''' CA2153: Do not catch corrupted state exceptions in general handlers
+    ''' </summary>
+    <DiagnosticAnalyzer(LanguageNames.VisualBasic)>
+    Public Class BasicDoNotCatchCorruptedStateExceptionsAnalyzer
+        Inherits DoNotCatchCorruptedStateExceptionsAnalyzer
+
+        Shared ReadOnly nodeKindsOfInterest As ImmutableArray(Of SyntaxKind) = ImmutableArray.Create(
+                SyntaxKind.SubBlock,
+                SyntaxKind.FunctionBlock,
+                SyntaxKind.OperatorBlock,
+                SyntaxKind.AddHandlerAccessorBlock,
+                SyntaxKind.RemoveHandlerAccessorBlock,
+                SyntaxKind.GetAccessorBlock,
+                SyntaxKind.SetAccessorBlock)
+
+
+        Protected Overrides Function GetAnalyzer(context As CompilationStartAnalysisContext, compilationTypes As CompilationMSInternalTypes) As Analyzer
+            Dim analyzer = New BasicAnalyzer(compilationTypes)
+            context.RegisterSyntaxNodeAction(AddressOf analyzer.AnalyzeNode, nodeKindsOfInterest)
+            Return analyzer
+        End Function
+
+        Private NotInheritable Class BasicAnalyzer
+            Inherits Analyzer
+
+            Public Sub New(compilationTypes As CompilationMSInternalTypes)
+                MyBase.New(compilationTypes)
+            End Sub
+
+            Private Function MayContainCatchNodesOfInterest(node As SyntaxNode) As Boolean
+
+                Dim kind = node.Kind()
+                Select Case kind
+                    Case SyntaxKind.MultiLineFunctionLambdaExpression
+                    Case SyntaxKind.MultiLineSubLambdaExpression
+                    Case SyntaxKind.SingleLineFunctionLambdaExpression
+                    Case SyntaxKind.SingleLineSubLambdaExpression
+                        ' for now there doesn't seem to have any way to annotate lambdas with attributes
+                        Return False
+                    Case Else
+                        Return True
+                End Select
+                Return True
+            End Function
+
+
+            Protected Overrides Sub CheckNode(methodNode As SyntaxNode, model As SemanticModel, reportDiagnostic As Action(Of Diagnostic))
```

Comment:
You could probably share a lot of code here between VB and C# analyzers. Define the base type to be an abstract generic type with type parameters `TCatchClauseSyntax`, `TThrowStatementSyntax` and abstract properties `TSyntaxKind CatchClauseKind {get;}` and `TSyntaxKind ThrowStatementKind {get;}` and almost all analysis code here can move to the base type?